streaming/util/shorthand.py

```diff
@@ -1,115 +1,371 @@
 
-__all__ = ['get_list_arg', 'bytes_to_int', 'number_abbrev_to_int']
+__all__ = [
+    'get_list_arg', 'get_str2str_arg', 'normalize_dec_bytes', 'normalize_bin_bytes',
+    'normalize_bytes', 'normalize_count', 'normalize_duration'
+]
 
 
-def get_list_arg(text: str) -> List[str]:
-    """Pass a list as a command-line flag.
+def get_list_arg(text: str, sep: str = ',') -> List[str]:
+    """Pass a list as a comma-delimited string.
 
     Args:
-        text (str): Text to split.
+        text (str): Text to parse.
 
     Returns:
-        List[str]: Splits, if any.
+        List[str]: List of items.
     """
-    return text.split(',') if text else []
+    if not text:
+        return []
 
+    return text.split(sep)
 
-def bytes_to_int(bytes_str: Union[int, str]) -> int:
-    """Convert human readable byte format to an integer.
+
+def get_str2str_arg(text: str, sep: str = ',', eq: str = '=') -> Dict[str, str]:
+    """Pass a dict as a comma- and equals-delimited string.
 
     Args:
-        bytes_str (Union[int, str]): Value to convert.
+        text (str): Text to parse.
+        sep (str): Separator text. Defaults to ``,``.
+        eq (str): Assignment text. Deffaults to ``=``.
+
+    Returns:
+        Dict[str, str]: Mapping of str to str.
+    """
+    if not text:
+        return {}
 
-    Raises:
-        ValueError: Invalid byte suffix.
+    ret = {}
+    parts = text.split(sep)
+    for part in parts:
+        key, val = part.split(eq)
+        if key in ret:
+            raise ValueError(f'Repeated key: {key} (text: {text}).')
+        ret[key] = val
+    return ret
+
+
+def _normalize_arg(text: str, units: Dict[str, int], to_type: type) -> Union[int, float]:
+    """Normalize a human-friendly unit string to number.
+
+    Args:
+        text (str): Human-friendly string.
+        units (Dict[str, Any]): Mapping of unit name to value.
+        to_type (Union[int, float]): The return type.
 
     Returns:
-        int: Integer value of bytes.
+        type: Computer-friendly number.
     """
-    #input is already an int
-    if isinstance(bytes_str, int) or isinstance(bytes_str, float):
-        return int(bytes_str)
-
-    units = {
-        'kb': 1024,
-        'mb': 1024**2,
-        'gb': 1024**3,
-        'tb': 1024**4,
-        'pb': 1024**5,
-        'eb': 1024**6,
-        'zb': 1024**7,
-        'yb': 1024**8,
-    }
-    # Convert a various byte types to an integer
-    for suffix in units:
-        bytes_str = bytes_str.lower().strip()
-        if bytes_str.lower().endswith(suffix):
-            try:
-                return int(float(bytes_str[0:-len(suffix)]) * units[suffix])
-            except ValueError:
-                raise ValueError(''.join([
-                    f'Unsupported value/suffix {bytes_str}. Supported suffix are ',
-                    f'{["b"] + list(units.keys())}.'
-                ]))
+    # Must be non-empty.
+    if not text:
+        raise ValueError(f'Text is empty.')
```

Comment:
Maybe a good idea to add the text in context to the facing parameter. Reading `'Text is empty` does not give me enough information on what is wrong with my param.

Reply:
How about `        raise ValueError(f'Attempting to normalize the empty strring.')`?

Reply:
Er, `        raise ValueError(f'Attempted to normalize an empty string to some value.')`